server/src/main/java/qonduit/store/SplitStore.java

```diff
@@ -0,0 +1,163 @@
+            Files.delete(p);
+        }
+    }
+
+    public void initialize() throws IOException {
+        if (Files.notExists(directory)) {
+            Files.createDirectories(directory);
+        } else if (!Files.isDirectory(directory)) {
+            throw new IOException(directory + " exists but is not a directory");
+        } else {
+            deleteExistingFiles();
+        }
+        task = EXECUTOR.scheduleWithFixedDelay(new RefreshTask(client), refreshIntervalSeconds, refreshIntervalSeconds,
+                TimeUnit.SECONDS);
+        LOG.info("SplitStore initialized");
+    }
+
+    public void writeSplitsForTable(AccumuloClient client, Map<String, String> tableMap, String tableName) {
+
+        if (tableName.startsWith(Namespace.ACCUMULO.name() + ".")) {
+            return;
+        }
+
+        LOG.debug("Getting splits for table: {}", tableName);
+        String tid = tableMap.get(tableName);
+        if (tid == null) {
+            // maybe a new table, try again
+            tableMap = client.tableOperations().tableIdMap();
+            tid = tableMap.get(tableName);
+            if (tid == null) {
+                // table deleted
+                LOG.info("{} is not in table map, maybe deleted?", tableName);
+                SPLIT_FILES.put(tableName, TABLE_DELETED_REF);
+                return;
+            }
+        }
+        final TableId tableId = TableId.of(tid);
+        try {
+            SortedSet<Text> splits = new TreeSet<>(client.tableOperations().listSplits(tableName));
+            Path tableSplitsFile = directory
+                    .resolve(tableName.replaceAll("\\.", "_") + "_splits_" + FILE_COUNTER.getAndIncrement() + ".rf");
+            LOG.debug("Writing splits for {} to {}", tableName, tableSplitsFile);
+            try (var writer = RFile.newWriter().to(tableSplitsFile.toString()).withTableProperties(FILE_PROPS)
+                    .build()) {
+                Text prev = null;
+                for (Text endRow : splits) {
+                    var extent = new KeyExtent(tableId, endRow, prev);
+                    var row = extent.toMetaRow();
+                    var key = new Key(row);
+                    key.setTimestamp(0);
+                    var encodedPrev = TabletColumnFamily.encodePrevEndRow(extent.prevEndRow());
+                    writer.append(key, encodedPrev);
+                    prev = endRow;
+                }
+                var extent = new KeyExtent(tableId, null, prev);
+                var row = extent.toMetaRow();
+                var key = new Key(row);
+                key.setTimestamp(0);
+                var encodedPrev = TabletColumnFamily.encodePrevEndRow(extent.prevEndRow());
+                writer.append(key, encodedPrev);
+            }
+            LOG.debug("Splits file {} complete for {}", tableSplitsFile, tableName);
+            SPLIT_FILES.computeIfAbsent(tableName, (t) -> new AtomicReference<Path>()).set(tableSplitsFile);
```

Comment:
I'm thinking I should remove the old file here. Might be safer to perform the check in SplitLookupOperation and delete the file if the RFile that was just read from is not the current reference.